### PR TITLE
feat[Collapse]: support classNames and styles

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [x] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 💡 Background and Solution

support classNames and styles

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     feat[Collapse]: support classNames and styles      |
| 🇨🇳 Chinese |   Collapse 支持语义化 classNames 和  styles  |

### CODE REVIEW

File: components/collapse/demo/basic.tsx [hunk 1]
```diff
@@ -13,6 +13,7 @@ const items: CollapseProps['items'] = [
     key: '1',
     label: 'This is panel header 1',
     children: <p>{text}</p>,
+    classNames: { header: 'header-custom', body: 'body-custom' },
```
Comment (zombieJ) [inline components/collapse/demo/basic.tsx:16]: 不要改 basic demo~
basic 都是尽量保持简单的，另外下面已经有 _semantic 的了，demo 就不用特地加了

File: components/collapse/demo/_semantic.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import type { CollapseProps } from 'antd';
+import { Collapse } from 'antd';
+
+import SemanticPreview from '../../../.dumi/components/SemanticPreview';
+import useLocale from '../../../.dumi/hooks/useLocale';
+
+const locales = {
+  cn: {
+    header: '设置面板头部区域',
```
Comment (zombieJ) [inline components/collapse/demo/_semantic.tsx:10]: 不用写`设置`，直接说 `头部区域`，下同

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore npm/@ant-design/tools@18.0.2
@SocketSecurity ignore npm/@biomejs/biome@1.8.3
@SocketSecurity ignore npm/puppeteer@22.15.0
@SocketSecurity ignore npm/sharp@0.33.5

File: components/collapse/demo/_semantic.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import type { CollapseProps } from 'antd';
+import { Collapse } from 'antd';
+
+import SemanticPreview from '../../../.dumi/components/SemanticPreview';
+import useLocale from '../../../.dumi/hooks/useLocale';
+
+const locales = {
+  cn: {
+    header: '头部区域',
```
Comment (zombieJ) [inline components/collapse/demo/_semantic.tsx:10]: 参考一下 Modal 的：https://ant-design.antgroup.com/components/modal-cn#semantic-dom

一般 xx 元素，偶尔加更多描述。英文的话也是 `element`。

**Review by zombieJ — APPROVED** [review]

**Pull request merged**